src/components/Posts/ComposeReply.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  Avatar,
+  Box,
+  Button,
+  Divider,
+  IconButton,
+  Stack,
+  TextField,
+} from "@mui/material";
+import { useState } from "react";
+import axios from "axios";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import EmojiEmotionsOutlinedIcon from "@mui/icons-material/EmojiEmotionsOutlined";
+import { appendPost } from "../../state/slices/postsSlice";
+
+type ComposeReplyProps = {
+  placeholder: string;
+};
+
+const styles = {
+  avatarIcon: { paddingRight: 1.5 },
+  compostPostContainer: {
+    justifyContent: "space-between",
+  },
+  textFieldContainer: {
+    width: "100%",
+    display: "flex",
+  },
+  textField: { paddingBottom: 2, paddingRight: 1 },
+  postActions: {
+    paddingLeft: 7.5,
+    paddingBottom: 2,
+  },
+  postButton: {
+    borderRadius: 5,
+    height: 35,
+    weight: 35,
+  },
+  topContainer: {
+    display: "flex",
+    padding: 2,
+    paddingBottom: 0,
+    justifyContent: "space-between",
+  },
+};
+
+const ComposeReply = ({ placeholder }: ComposeReplyProps) => {
+  const [postTextContent, setPostTextContent] = useState("");
+  const [focusReply, setFocusReply] = useState(false);
+  const user = useAppSelector((state) => state.user);
+  const post = useAppSelector((state) => state.post);
+  const dispatch = useAppDispatch();
+
+  const onSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    try {
+      const textContent = postTextContent;
+      const reply = await axios.post(
+        "http://localhost:3001/api/posts/postReply",
+        {
+          userId: user.userId,
+          parentPostId: post.postId,
+          textContent,
+        }
+      );
+      setPostTextContent("");
+      dispatch(
+        appendPost({
+          ...reply.data,
+          username: user.username,
+          displayName: user.displayName,
+          numberOfLikes: 0,
+        })
+      );
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <form onSubmit={onSubmit}>
+      <Box sx={styles.compostPostContainer}>
+        <Box sx={styles.topContainer}>
+          <Box sx={styles.avatarIcon}>
+            <Avatar />
+          </Box>
+          <Box sx={styles.textFieldContainer}>
+            <TextField
+              fullWidth
+              hiddenLabel
+              id="standard-multiline-static"
+              multiline
+              onChange={(e) => setPostTextContent(e.target.value)}
+              placeholder={placeholder}
+              sx={styles.textField}
+              value={postTextContent}
+              variant="standard"
+              onFocus={() => setFocusReply(true)}
+              onBlur={() => setFocusReply(false)}
```

Comment:
Adding an `onBlur` still makes the component act weird, cause when you try to click on the icons they disappear right away. This happens because the `onBlur` is only on the `TextField`, and the icons are outside of the `TextField`

I'm not really sure how to solve at first glance. Wish we could make the `onBlur` on the parent container instead.. If you have trouble coming up with something we can pair program and try stuff together

Reply:
Yeah this was my temporary solution tbh, I don't like it either 

I can't really update it because like you said I don't have access to it since the action is in a separate component to the Compose Reply component

Reply:
As discussed, we'll revert this to the way you originally had it and make a new ticket to more fully flesh out the behavior of the buttons